src/test/regress/expected/foreign_key_to_reference_table.out

```diff
@@ -0,0 +1,785 @@
+INSERT INTO referenced_table SELECT x, x+1 FROM generate_series(1,1000) AS f(x);
+INSERT INTO referencing_table SELECT x, x+1 FROM generate_series(1,999) AS f(x);
+INSERT INTO referenced_table VALUES (-1,-1);
+DELETE FROM referenced_table WHERE test_column > 800;
+SELECT count(*) FROM referencing_table WHERE ref_id = -1;
+ count 
+-------
+   200
+(1 row)
+
+DROP TABLE referencing_table;
+DROP TABLE referenced_table;
+-- foreign key as composite key 
+CREATE TYPE composite AS (key1 int, key2 int);
+SELECT run_command_on_workers($$CREATE TYPE fkey_reference_table.composite AS (key1 int, key2 int)$$);
+      run_command_on_workers       
+-----------------------------------
+ (localhost,57637,t,"CREATE TYPE")
+ (localhost,57638,t,"CREATE TYPE")
+(2 rows)
+
+CREATE TABLE referenced_table(test_column composite, PRIMARY KEY(test_column));
+CREATE TABLE referencing_table(id int, referencing_composite composite);
+SELECT create_reference_table('referenced_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('referencing_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+ALTER TABLE referencing_table ADD CONSTRAINT fkey_ref FOREIGN KEY (referencing_composite) REFERENCES referenced_table(test_column) ON DELETE CASCADE;
+INSERT INTO referenced_table SELECT (x, x+1)::composite FROM generate_series(1,1000) AS f(x);
+INSERT INTO referencing_table SELECT x, (x+1, x+2)::composite FROM generate_series(1,999) AS f(x);
+DELETE FROM referenced_table WHERE (test_column).key1 > 900;
+SELECT count(*) FROM referencing_table;
+ count 
+-------
+   899
+(1 row)
+
+DROP TABLE referenced_table CASCADE;
+NOTICE:  drop cascades to constraint fkey_ref on table referencing_table
+DROP TABLE referencing_table CASCADE;
+-- Serial column type works properly when it is in the target of foreign constraint
+CREATE TABLE referenced_table(test_column SERIAL PRIMARY KEY, test_column2 int);
+CREATE TABLE referencing_table(id int, ref_id int);
+SELECT create_reference_table('referenced_table');
+ create_reference_table 
+------------------------
+ 
+(1 row)
+
+SELECT create_distributed_table('referencing_table', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+ALTER TABLE referencing_table ADD CONSTRAINT fkey_ref FOREIGN KEY (ref_id) REFERENCES referenced_table(test_column) ON DELETE CASCADE;
+INSERT INTO referenced_table(test_column2) SELECT x%20 FROM generate_series(1,1000) AS f(x);
+INSERT INTO referencing_table SELECT x, x+1 FROM generate_series(1,999) AS f(x);
+DELETE FROM referenced_table WHERE test_column2 > 10;
+SELECT count(*) FROM referencing_table;
+ count 
+-------
+   549
+(1 row)
+
+DROP TABLE referenced_table CASCADE;
+NOTICE:  drop cascades to constraint fkey_ref on table referencing_table
+DROP TABLE referencing_table CASCADE;
+-- Serial column type works properly when it is in the source of foreign constraint
```

Comment:
It'd be great if you can expand the comments per test block that we execute queries. Here we say that it works properly, but, in the end, the query fails. So, as a reader, I was initially a bit surprised to see that. For example, for this test, I'd write the following:
```
-- In the following test, we'll use a SERIAL column as the referencing column
-- in the foreign constraint. We'll first show that the values that exist 
-- in the referenced tables are successfully generated by the serial column
-- and inserted to the distributed table. However,  if the values that are generated 
-- by serial column do not exist on the referenced table, the query fails.
```